### ISSUE TITLE
language/go: update-repos adds sums to go.sum with Go 1.16

### ISSUE THREAD

**Issue by jayconrod** [bazelbuild/bazel-gazelle#990]
### What version of gazelle are you using?

e4496b956eb46bdf8c9bf95b8d1d85e3a086c4be

### What version of rules_go are you using?

46c56d280eeee5b6e71b710f9afb6ab49ffcdf51

### What version of Bazel are you using?

4.0.0

### Does this issue reproduce with the latest releases of all the above?

yes

### What operating system and processor architecture are you using?

darwin amd64

### What did you do?

* Remove module content hashes from `go.sum` (those without the `/go.mod` suffix)
* Run `gazelle update-repos -from_file=go.mod`.

### What did you expect to see?

Gazelle does not modify `go.sum` or `go.mod`.

### What did you see instead?

Gazelle adds missing sums to `go.sum`.

This happens because Gazelle runs `go mod download` to find missing sums. In Go 1.16, `go mod download` adds module content sums to `go.sum`. Previously, `go mod download` only updated `go.sum` after loading the build list, so it would only add `/go.mod` hashes. That was probably a bug before we didn't notice.

**Comment by gstroz** [comment]
I feel like I'm seeing something that is related to this but appears to be the opposite issue. Gazelle is failing to update the `go.sum` file because `go mod` is in read only mode.

I created a sample repo which demonstrates the issue. I confirmed that it's also happening on the same version of gazelle mentioned here. I am running rules go v0.25.1. Bazel 4.0.0

https://github.com/gstroz/gazelle-issue

Run through bazel:

```
bazel run //:gazelle -- update-repos -from_file=go.mod -build_file_generation auto -build_file_proto_mode disable
INFO: Analyzed target //:gazelle (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 0.141s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/gazelle -bazel_run update-repos '-from_file=go.mod' -build_file_generation auto -build_file_proto_modeINFO: Build completed successfully, 1 total action
gazelle: running '/home/gstroz/.cache/bazel/_bazel_gstroz/10ac875636ae7271d155e6ad32b80492/external/go_sdk/bin/go /home/gstroz/.cache/bazel/_bazel_gstroz/10ac875636ae7271d155e6ad32b80492/external/go_sdk/bin/go list -mod=readonly -m -json all': go: github.com/imroc/req@v0.3.0: missing go.sum entry; to add it:
	go mod download github.com/imroc/req
```


running standalone gazelle e4496b9
```
./gazelle update-repos -from_file=go.mod
gazelle: running '/usr/local/go/bin/go go list -mod=readonly -m -json all': go: updates to go.sum needed, disabled by -mod=readonly
```

**Comment by peplin** [comment]
I'm collaborating with @gstroz on this issue, and I just found that the problem is resolved if I use v0.22.3 with #912 reverted.

**Comment by derekperkins** [comment]
Is this resolved in v0.23.0?

**Comment by fishy** [comment]
> Is this resolved in v0.23.0?

doesn't seem so.

**Comment by fishy** [comment]
Is there any plan to cut a release with the fix?
